app/validators/enabled_dynamic_field/open_to_all_projects_validator.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class EnabledDynamicField::OpenToAllProjectsValidator < ActiveModel::Validator
+  def validate(enabled_dynamic_field)
+    return if enabled_dynamic_field.open_to_all_projects || enabled_dynamic_field.project&.is_primary
+    enabled_dynamic_field.errors[:open_to_all_projects] << "Only primary projects have exclusively enabled dynamic fields"
```

Comment:
@barmintor Maybe worth changing this error message to align more with the field name?  Something like "Option 'Open to all projects' can only be enabled for fields in primary projects."  Might help with troubleshooting if we encounter this validation error in the future.